### PR TITLE
run npm update on fab deploy

### PR DESCRIPTION
**Pull request by dannyroberts**

### CODE REVIEW

**Comment by benrudolph** [comment]
i don't think this is a big deal since we have so few npm packages, but we could potentially do the same optimization trick we do with bower (copy the folder over if it exists)

**Comment by benrudolph** [comment]
see: https://github.com/dimagi/commcare-hq/blob/fab-npm/fab/fabfile.py#L705

**Comment by dannyroberts** [comment]
oh nice, thanks, I'll add that

**Comment by dannyroberts** [comment]
@benrudolph This finally worked on staging; this'll have to be merged before deploy (or revert https://github.com/dimagi/commcare-hq/pull/10028)

**Comment by NoahCarnahan** [comment]
:+1:

File: fab/fabfile.py [hunk 1]
```diff
@@ -1057,6 +1068,15 @@ def _bower_install(use_current_release=False):
         sudo('bower update --production --config.interactive=false')
 
 
+@parallel
+@roles(ROLES_DJANGO)
+def _npm_install():
+    with cd(env.code_root):
+        sudo('npm prune --production')
+        sudo('npm install --production')
```
Comment (benrudolph) [inline fab/fabfile.py]: technically this isn't necessary

**Comment by benrudolph** [comment]
:+1: though should remove command to not add cruft to deploy time

File: fab/fabfile.py (hunk 1 above)
Comment (benrudolph) [inline fab/fabfile.py]: it seems like in your commit history you found differently

Comment (benrudolph) [inline fab/fabfile.py]: (locally works fine to just do `update` for me)

Comment (dannyroberts) [inline fab/fabfile.py]: Oh sorry, I added it back because it *was* necessary. For npm 1.x, install gets js-xpath but update doesn't.

**Pull request merged**